packages/server/lib/config.coffee

```diff
@@ -327,9 +340,24 @@ module.exports = {
     ## and change the resolved values of cfg
     ## to point to the plugin
     setResolvedOn(cfg.resolved, diffs)
+    debug("resolved config object %o", cfg.resolved)
 
     ## merge cfg into overrides
-    _.defaultsDeep(diffs, cfg)
+    resolved = _.defaultsDeep(diffs, cfg)
+
+    ## Take a special care with some system properties the
+    ## user should be able to modify. For example, if the plugins file
+    ## modifes detected browsers, the _.defaultsDeep merges it back
+    ## and we need to use the user's value
+
+    # do not allow user to delete "browsers" list - otherwise how to run tests?
+    if Array.isArray(userBrowserList)
+      debug("using user supplied list of browsers %o", userBrowserList)
+      resolved.browsers = userBrowserList
```

Comment:
Doesn't line 346 copy `diffs.browsers` in to `resolved` unconditionally?

Reply:
right, the default behavior is to use all browsers. But if the user has returned a list of browsers, we need to resolve with that list